AIAgent/common/game.py

```diff
@@ -88,14 +89,9 @@ class MoveReward:
     ForCoverage: int
     ForVisitedInstructions: int
 
-    def __eq__(self, __o) -> bool:
-        if type(self) != type(__o):
-            raise RuntimeError(f"Can't compare {type(self)} with {type(__o)}")
```

Comment:
Why it is removed?

Reply:
1. The method was written incorrectly
 <details>
  <summary>Log</summary>

  ```python
  Traceback (most recent call last):
  File "/root/programming/PySymGym/AIAgent/common/game.py", line 124, in <module>
    a == b
  File "/root/programming/PySymGym/AIAgent/common/game.py", line 98, in __eq__
    return self == __o
           ^^^^^^^^^^^
  File "/root/programming/PySymGym/AIAgent/common/game.py", line 98, in __eq__
    return self == __o
           ^^^^^^^^^^^
  File "/root/programming/PySymGym/AIAgent/common/game.py", line 98, in __eq__
    return self == __o
           ^^^^^^^^^^^
  [Previous line repeated 496 more times]
RecursionError: maximum recursion depth exceeded
```
  
</details>

2. I think the default behavior of ___eq__ (dataclass implementation) was implied, so I decided to remove the overriding